src/test/regress/README.md

```diff
@@ -81,6 +81,35 @@ Adding a new test file is quite simple:
 4. Check that the output is as expected
 5. Copy the `.out` file from `results` to `expected`
 
+### Adding an alternative test output
+
+Sometimes the test output changes because we run tests in different configurations.
+The most common example is an output that changes in different Postgres versions.
+If we don't care about the change, we try to add [a normalization rule](https://github.com/citusdata/citus/blob/main/ci/README.md#normalize_expectedsh), or find
+ways to avoid the output altogether.
+
+If we actually care about the output in all its forms, we add an alternative test
+output. In order to maintain a clean test suite, make sure to explain in the test
+header the reason why it has an alternative output, and when we can drop the
+alternative output in the future.
+
+For example, the header of [multi_insert_select.sql](https://github.com/citusdata/citus/blob/main/src/test/regress/sql/multi_insert_select.sql) is as follows:
+
+```sql
+--
+-- MULTI_INSERT_SELECT
+--
+-- This test file has an alternative output because of the change in the
+-- display of SQL-standard function's arguments in INSERT/SELECT in PG15.
+-- The alternative output can be deleted when we drop support for PG14
+--
+```
+Including important keywords, like "PG14", "PG15", "alternative output" will
+help cleaning up in the future.
+
+This is just an example - there are no strict rules here. Simply try to add the
+alternative output in a readable way for other developers.
```

Comment:
These last two sentences kind of diminish your point. I think everything you explained here is very important and people are already not going to follow rules 😄

Reply:
Makes sense. I start off strong and get softer along the way 😆